lib/providers/token.js

```diff
@@ -202,6 +202,10 @@ class TokenProvider {
    * @private
    */
   _post(payload) {
+    const payloadStr = JSON.stringify(payload);
+
+    this._warden.headers['Content-Length'] = payloadStr.length;
```

Comment:
Do we want to clone `this._warden` and modify the cloned copy instead? I'm wondering about the case where you POST two separate payloads with different content lengths and one resolves before the other.